src/Repository/EntitySpecificationRepositoryTrait.php

```diff
@@ -176,8 +176,16 @@ public function getQueryBuilder($specification, ?string $alias = null): QueryBui
      */
     public function iterate($specification, ?ResultModifier $modifier = null): \Traversable
     {
-        foreach ($this->getQuery($specification, $modifier)->iterate() as $key => $row) {
-            yield $key => current($row);
+        $query = $this->getQuery($specification, $modifier);
+
+        if (method_exists($query, 'toIterable')) {
+            foreach ($query->toIterable() as $key => $row) {
+                yield $key => $row;
+            }
```

Comment:
Can this be replaced with `yield from`?
```suggestion
            yield from $query->toIterable();
```

Reply:
Yes. You're right. This would be better.